packages/lit-query/src/__tests__/TodoComponent.ts

```diff
@@ -0,0 +1,55 @@
+import { LitElement, html } from 'lit'
+import { property } from 'lit/decorators.js'
+import { vi } from 'vitest'
+import { QueryController } from '../QueryController'
+
+export const A_TODO_ID = 1
+export const A_TODO = {
+  userId: 1,
+  id: 1,
+  title: 'My first todo',
+  completed: false,
+}
+export const ANOTHER_TODO_ID = 2
+export const ANOTHER_TODO = {
+  userId: 1,
+  id: 2,
+  title: 'My second todo',
+  completed: false,
+}
+
+export type Todo = {
+  readonly userId: number
+  readonly id: number
+  readonly title: string
+  readonly completed: boolean
+}
+
+export const getTodoById = vi.fn<[todoId: number], Promise<Todo>>()
+
+export class ReadOneTodoComponent extends LitElement {
+  @property({ type: Number })
+  todoId = 1
+
+  todoQuery = new QueryController(this, () => ({
```

Comment:
The ergonomics of this are similar to Lit's [Async Tasks](https://lit.dev/docs/data/task/). Great work!